File: lms_front/lib/features/shared/data/models/course_related/course.dart
import 'package:json_annotation/json_annotation.dart';

part 'course.g.dart';

@JsonSerializable()
class Course {
  final String id;
  final String title;

  /// Optional start date of the course
  @JsonKey(name: 'start_ts', fromJson: DateTime.parse)
  final DateTime? startDate;

  /// Optional end date of the course
  @JsonKey(name: 'end_ts', fromJson: DateTime.parse)
  final DateTime? endDate;

  final String description;

  @JsonKey(name: 'author_id')
  final String authorId;

  Course({
    required this.id,
    required this.title,
    required this.startDate,
    required this.endDate,
    required this.description,
    required this.authorId,
  });

  factory Course.fromJson(Map<String, dynamic> json) => _$CourseFromJson(json);
  Map<String, dynamic> toJson() => _$CourseToJson(this);
}

File: service/pg_service_template/src/models/courses/course.cpp
#include "course.hpp"

namespace lms_service 
{

void data_pending(
    userver::formats::json::ValueBuilder& item, const Course& course) 
    {
      item["id"] = course.course_id;
      item["title"] = course.title;
      item["description"] = course.description;
      item["author_id"] = course.author_id;

      if (course.start_ts.has_value())
        item["start_ts"] = course.start_ts.value();
      if (course.end_ts.has_value())
        item["end_ts"] = course.end_ts.value();
    }
}  // namespace lms_service